static-assets/components/cstudio-common/common-api.js

```diff
@@ -3379,6 +3380,27 @@ var nodeOpen = false;
 
                 YConnect.asyncRequest('POST', this.createServiceUri(serviceUrl), serviceCallback);
             },
+
+            syncFromRepo: function(site, callback){
+                var serviceUrl = this.syncRepoServiceUrl;
+                var postData = "site_id=" + site;
+
+                var serviceCallback = {
+                    success: function(jsonResponse) {
+                        var results = {};
+                        if(jsonResponse.responseText != "") {
+                            results = eval("(" + jsonResponse.responseText + ")");
```

Comment:
Is it possible to use JSON.parse?  I see that in some places and eval in others.